class-two-factor-core.php

```diff
@@ -1925,4 +1927,30 @@ public static function rememberme() {
 
 		return (bool) apply_filters( 'two_factor_rememberme', $rememberme );
 	}
+
+	/**
+	 * Sync the Two-Factor session data from the current session to newly created sessions.
+	 *
+	 * This is required as WordPress creates a new session when the user password is changed.
+	 *
+	 * @see https://core.trac.wordpress.org/ticket/58427
+	 *
+	 * @param array $session The Session information.
+	 * @param int   $user_id The User ID for the session.
+	 * @return array
+	 */
+	public static function filter_session_information( $session, $user_id ) {
```

Comment:
A unit test for this would be ideal if it's not gonna be a pain to setup.

Reply:
added in 52c1c0d